packages/react/docs/guides/onboarding.md

```diff
@@ -0,0 +1,259 @@
+# Onboarding
+
+The purpose of this document is to help new developers quickly and easily onboard to this project and
+pick up the styles and practices we use to make this a coherent code base.
+
+## Coding practices
+
+Below you will find major items we strive to provide in all of our code and try to check for in all
+code reviews.
+
+### Prop naming conventions
+
+When naming props all names should be in camelCase, events should be prefixed with `on` (onClick), and booleans should be prefixed with `is` or `has` (isActive, hasLabel).
+
+### Localization
+
+All non-falsy translatable strings should be passed in an `i18n` prop and the defaultProps should include
+an English default. If you have multiple default, but only want to override some of them, you can use
+the `useMerged` helper hook to merge the objects together.
+
+```jsx
+// component
+const defaultProps = {
+  onClick: undefined,
+  i18n: {
+    labelText: 'button label',
+  },
+};
+
+const Button = ({ i18n, onClick, children }) => {
+  const mergedI18n = useMerged(defaultProps.i18n, i18n);
+  return (
+    <button aria-label={mergedI18n.labelText} onClick={onClick}>
+      {children}
+    </button>
+  );
+};
+
+// usage
+<Button i18n={{ labelText: 'A fancy button' }}>Fancy button</Button>;
+```
+
+### Adding data-testid
+
+We want to make sure our components have dependable way to test all major functionality, so we provide
+this to our consumers with `data-testid` props on important parts of the component. For example,
+a button component could look like this:
```

Comment:
Something about also adding test-id to the subcomponents and how that is done?

Reply:
Good idea. Added an example.